scale/job/test/messages/test_process_job_input.py

```diff
@@ -364,3 +364,98 @@ def test_execute_with_recipe(self):
         # Make sure job input file models are unchanged
         job_input_files = JobInputFile.objects.filter(job_id=job_2.id)
         self.assertEqual(len(job_input_files), 4)
+
+    def test_execute_with_recipe_missing_output(self):
+        """Tests calling ProcessJobInput.execute() when a job has to get its data from its recipe but the previous job didn't provide it's output"""
+
+        workspace = storage_test_utils.create_workspace()
+        file_1 = storage_test_utils.create_file(workspace=workspace, file_size=104857600.0)
+        file_2 = storage_test_utils.create_file(workspace=workspace, file_size=987654321.0)
+        file_3 = storage_test_utils.create_file(workspace=workspace, file_size=65456.0)
+        file_4 = storage_test_utils.create_file(workspace=workspace, file_size=24564165456.0)
+        manifest_1 = {
+            'seedVersion': '1.0.0',
+            'job': {
+                'name': 'job-a',
+                'jobVersion': '1.0.0',
+                'packageVersion': '1.0.0',
+                'title': '',
+                'description': '',
+                'maintainer': {
+                    'name': 'John Doe',
+                    'email': 'jdoe@example.com'
+                },
+                'timeout': 10,
+                'interface': {
+                    'command': '',
+                    'inputs': {'files': [], 'json': []},
+                    'outputs': {
+                        'files': [{'name': 'OUTPUT_A', 'pattern': '*.png'}]
+                    }
+                }
+            }
+        }
+        job_type_1 = job_test_utils.create_job_type(interface=manifest_1)
+        manifest_2 = {
+            'seedVersion': '1.0.0',
+            'job': {
+                'name': 'job-b',
+                'jobVersion': '1.0.0',
+                'packageVersion': '1.0.0',
+                'title': '',
+                'description': '',
+                'maintainer': {
+                    'name': 'John Doe',
+                    'email': 'jdoe@example.com'
+                },
+                'timeout': 10,
+                'interface': {
+                    'command': '',
+                    'inputs': {'files': [{'name': 'INPUT_B', 'multiple': True}]},
+                    'outputs': {
+                        'files': [{'name': 'OUTPUT_B', 'pattern': '*.png'}]
+                    }
+                }
+            }
+        }
+        job_type_2 = job_test_utils.create_job_type(interface=manifest_2)
```

Comment:
Same comment as above.

Reply:
Good catch.  I just need to handle failing v5 jobs without required output.  I went down a rabbit hole making sure we were checking this for seed jobs and realized at the end we were already doing that correctly.  So now I'm going down the path for legacy jobs.